spark/src/test/scala/org/apache/spark/sql/delta/schema/SchemaUtilsSuite.scala

```diff
@@ -2569,6 +2573,244 @@ class SchemaUtilsSuite extends QueryTest
     assert(mergedSchema === expected)
   }
 
+  test("typeWideningMode - byte->short->int is always allowed") {
+    import TypeWideningMode._
+    val narrow = new StructType()
+      .add("a", ByteType)
+      .add("b", ByteType)
+      .add("c", ShortType)
+      .add("s", new StructType().add("x", ByteType))
+      .add("m", MapType(ByteType, ShortType))
+      .add("ar", ArrayType(ByteType))
+
+    val wide = new StructType()
+      .add("a", ShortType)
+      .add("b", IntegerType)
+      .add("c", IntegerType)
+      .add("s", new StructType().add("x", IntegerType))
+      .add("m", MapType(ShortType, IntegerType))
+      .add("ar", ArrayType(IntegerType))
+
+    for (typeWideningMode <- Seq(
+        NoTypeWidening,
```

Comment:
Does it make sense to create a list of all modes? I think we would use it more than once

Reply:
:+1:, I added it